File: taskmanager/src/main/java/com/example/alexeyglushkov/taskmanager/loader/http/HttpBytesTransport.java
package com.example.alexeyglushkov.taskmanager.loader.http;

import com.example.alexeyglushkov.streamlib.data_readers_and_writers.ByteArrayReader;

import java.net.HttpURLConnection;

/**
 * Created by alexeyglushkov on 26.11.15.
 */
public class HttpBytesTransport<T> extends HttpTaskTransport<T> {
    protected ByteArrayReader<T> byteArrayReader;

    public HttpBytesTransport(HttpURLConnectionProvider provider, final HTTPConnectionBytesReader<T> handler) {
        super(provider, null);

        byteArrayReader = new ByteArrayReader<>(handler, true);
        setStreamReader(new HTTPConnectionStreamReaderAdaptor<T>(byteArrayReader){
            @Override
            public void handleConnectionResponse(HttpURLConnection connection) {
                if (handler != null) {
                    handler.handleConnectionResponse(connection);
                }
            }
        });
    }
}

File: taskmanager_httptask/src/main/java/com/aglushkov/taskmanager_http/loader/http/HttpBytesTransport.kt
package com.aglushkov.taskmanager_http.loader.http

import com.example.alexeyglushkov.streamlib.data_readers_and_writers.ByteArrayReader
import java.net.HttpURLConnection

/**
 * Created by alexeyglushkov on 26.11.15.
 */
open class HttpBytesTransport<T>(provider: HttpURLConnectionProvider,
                                 handler: HTTPConnectionBytesReader<T>) : HttpTaskTransport<T>(provider, NullHTTPConnectionStreamReader<T>()) {
    protected var byteArrayReader: ByteArrayReader<T>

    init {
        byteArrayReader = ByteArrayReader(handler, true)
        streamReader = object : HTTPConnectionStreamReaderAdaptor<T>(byteArrayReader) {
            override fun handleConnectionResponse(connection: HttpURLConnection) {
                handler.handleConnectionResponse(connection)
            }
        }
    }
}